fe/fe-core/src/main/java/org/apache/doris/nereids/properties/FuncDeps.java

```diff
@@ -100,7 +100,11 @@ public Set<Set<Slot>> eliminateDeps(Set<Set<Slot>> slots) {
         Set<Set<Slot>> minSlotSet = slots;
         List<Set<Set<Slot>>> reduceSlotSets = new ArrayList<>();
         reduceSlotSets.add(slots);
-        while (!reduceSlotSets.isEmpty()) {
+        // To avoid memory usage due to multiple iterations,
+        // we set a maximum number of loop iterations.
+        int count = 0;
+        while (!reduceSlotSets.isEmpty() && count < 100) {
+            count += 1;
```

Comment:
could u add an example to explain why need more than 100 loops to reduce slots?

Reply:
I haven't found the real reason yet; this is just a temporary solution to avoid oom in of the user cases in version 3.0.